trin-core/src/portalnet/overlay_service.rs

```diff
@@ -431,10 +478,53 @@ where
                         self.peers_to_ping.insert(node_id);
                     }
                 }
+                query_event = OverlayService::<TContentKey, TMetric, TValidator>::query_event_poll(&mut self.queries) => {
+                    match query_event {
+                        // Send a FINDNODES on behalf of the query.
+                        FindNodeQueryEvent::Waiting(query_id, node_id, request) => {
+
+                            // Look up the node's ENR.
+                            if let Some(enr) = self.find_enr(&node_id) {
+
+                                let request = OverlayRequest::new(
+                                    request,
+                                    RequestDirection::Outgoing {
+                                        destination: enr.clone()
+                                    },
+                                    None,
+                                    Some(query_id),
+                                );
+                                let _ = self.request_tx.send(request);
+
+                            } else {
+                                error!("[{:?}] Unable to send FINDNODES to unknown ENR with node ID {}",
+                                         self.protocol, node_id);
+                            }
+
+                        }
+                        // Query has ended.
+                        FindNodeQueryEvent::Finished(query_id, query) | FindNodeQueryEvent::TimedOut(query_id, query) => {
+                            let result = query.into_result();
+                            // Obtain the ENRs for the resulting nodes.
+                            let mut found_enrs = Vec::new();
+                            for node_id in result.into_iter() {
+                                if let Some(enr) = self.find_enr(&node_id) {
+                                    found_enrs.push(enr);
+                                } else if let Some(enr) = self.discovery.discv5.find_enr(&node_id) {
+                                    found_enrs.push(enr);
+                                }
+                                else {
+                                    warn!("ENR not present in queries results.");
+                                }
+                            }
```

Comment:
Yea, I removed query_info in an apparent brainfart end of last week.

I think that this was still working though, because `find_enr` could still get the ENRs from the routing table.